pass some tls options in order to use "secure mirrors"

toenseo opened the pull request:
Hi,

in our environment the build system is only allowed to contact one specific mirror server for download.
This server is secured using client authentification, so we have to pass a certificate.
I added some code to pass environment variables to the request options and would be very excited if this code could be merged into the main package.

Any review and improvement comments are very welcome!

Thank you and have a nice day!
Oliver

toenseo commented:
Thanks for all the helpful comments. I hope I could completely meet the requirements now?

ralphtheninja commented:
+1 from me if you squash the commits (interactive rebase), want to get feedback from @mafintosh before merging :)

toenseo committed:
pass some tls options in order to use secure mirrors

commit 8f8c8e2d54bd6228125ffe76a4893ac85721db4c
Author: otoense <oliver@toense.de>
Date:   Tue Nov 10 14:01:17 2015 +0100

    Revert "add intellij files to gitignore"

    This reverts commit 34b98f45998af9bb20ed566afde7d2edcd6a870a.

commit 0ce54037696c3951612e6a1c1910bf2bc1178d1c
Author: otoense <oliver@toense.de>
Date:   Tue Nov 10 13:31:08 2015 +0100

    meet 'standard' code style

commit 8fbe40546e401c3562df13c0847f00af8d20a653
Author: otoense <oliver@toense.de>
Date:   Tue Nov 10 12:34:30 2015 +0100

    fix readme

commit e954663b1e5ac1bc98ef19c3beb7c96239cc1663
Author: otoense <oliver@toense.de>
Date:   Tue Nov 10 12:32:04 2015 +0100

    extend readme by 'secure mirrors' section

commit 39200eb3013282a7e37bf3601fbd281dde3dff33
Author: otoense <oliver@toense.de>
Date:   Tue Nov 10 09:48:10 2015 +0100

    add https/tls configuration

commit 34b98f45998af9bb20ed566afde7d2edcd6a870a
Author: otoense <oliver@toense.de>
Date:   Tue Nov 10 09:47:41 2015 +0100

    add intellij files to gitignore

toenseo commented:
Thanks for your+1 :-) I hopefully squashed the commits now as expected.

ralphtheninja commented:
@otoense It seems you didn't. The old commits are still in the history. No rush, take your time :)

toenseo commented:
Oh, you are so right. Sorry. Had to push --force but seems ok now, doesn't it?

ralphtheninja commented:
Yup. Now you have +1 from me :rocket:

toenseo commented:
@ralphtheninja Thank you very much for your patience! I've learned a lot today :-)

mafintosh commented:
Great! Released in 2.2.0

toenseo commented:
Thank you very much for this quick merge! Have a nice day!

ralphtheninja commented:
:)
